### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by yosaddis**
# In this feature I added the back end of the project

Git initialization for version control.
PostgreSQL database setup.
Creation of a 'messages' table with 5 different greetings.
Addition of an API endpoint for selecting a random greeting.

### CODE REVIEW

File: app/controllers/message_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class MessageController < ApplicationController
+  def index
+    @message = Message.order('RANDOM()').first
+    render json: { greeting: @message.greeting }
+  end
```
Comment (Davidosky007) [inline app/controllers/message_controller.rb:5]: * I get CORS errors in my console when I try to run your app, Kindly [Install the CORS gem and follow the configurations to fix this error](https://github.com/cyu/rack-cors) 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @yosaddis 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well-written PR 👍 
* Good looking readme 👍 
* Link to backend API provided 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get 'message/index', to: 'message#index'
   # Defines the root path route ("/")
   # root "articles#index"
```
Comment (FrederickMih) [inline config/routes.rb:5]: - [Optional] API versioning is important: This provide a way to evolve the API over time without breaking existing clients app.

Comment (FrederickMih) [inline config/routes.rb:5]: - [Optional] Indicating the root routes is important-- This prevents the default Rails set up page from displaying :+1: 
Screenshot from 2023-10-08 05-25-39

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get 'message/index', to: 'message#index'
```
Comment (FrederickMih) [inline config/routes.rb:3]: - [ ] Opening `http://localhost:3000/message/index` on the browser produces the error on the screenshot below hence your two apps are not working together. Let's kindly fix :wrench: the error
image

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @yosaddis, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Highlights :green_circle: 

- Great job fixing the CORS issues raised by the previous reviewer :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,9 @@
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+Message.create('Hello')
+Message.create('Hola')
+Message.create('Bonjour')
+Message.create('Ciao')
```
Comment (Shedrack-Sunday) [inline db/seeds.rb]: - [ ] To fix the error in your code, kindly declare the entry like this:

### Sample:
```
# db/seeds.rb

Message.create(greeting: "Hello, World!")
```
-- The lack of `greeting` method is the reason of the error

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @yosaddis 👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Rails features ✔️
- Clean use of class calls ✔️  

# Required Changes  🌵  

- [x] The changes required are minor but very important so we avoid bugs in the application.

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by topeogunleye — APPROVED** [review]
### Hi @yosaddis, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 

✅ Great job addressing the feedback from the previous review!
✅ Nice correction in the `db/seeds.rb` by declaring the entry as suggested.
✅ By adding the `greeting: "Hello, World!"` in the `Message.create`, you've successfully fixed the error related to the missing `greeting` method. Well done! 👍

Keep up the good work! 💪 
## [Optional] suggestions:
- For future pull requests, please notice that your branch name `initial-setup` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-initial-setup`. This makes it easier to track the changes and the purpose of each branch. 

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**